DAQ_FW/lib/Log/Log.h

```diff
@@ -17,16 +15,40 @@ namespace Logger
 
     void Start();
 
-    void Verbose(const char *msg);
-
-    void Trace(const char *msg);
-
-    void Info(const char *msg);
-
-    void Warning(const char *msg);
-
-    void Error(const char *msg);
-
-    void Fatal(const char *msg);
+    template <class T, typename... Args>
+    void Verbose(T msg, Args... args)
+    {
+        Log.verboseln(msg, args...);
+    }
+
+    template <class T, typename... Args>
+    void Trace(T msg, Args... args)
+    {
+        Log.traceln(msg, args...);
+    }
+
+    template <class T, typename... Args>
+    void Notice(T msg, Args... args)
+    {
+        Log.noticeln(msg, args...);
```

Comment:
It's not immediately obvious when I should be using `Trace` and `Notice` over the other functions. What exactly do they do?

Reply:
`Trace` and `Notice` call the upper level log messages, these can be filtered out by setting the `loglevel` at compile or runtime to a lower value. 

I used the same names as the Arduino-Log library, but we can change them to whatever we need. It is up to the user how these are used, the current log level heirchy in the library is this:
```
* 0 - LOG_LEVEL_SILENT     no output 
* 1 - LOG_LEVEL_FATAL      fatal errors 
* 2 - LOG_LEVEL_ERROR      all errors  
* 3 - LOG_LEVEL_WARNING    errors, and warnings 
* 4 - LOG_LEVEL_NOTICE     errors, warnings and notices 
* 5 - LOG_LEVEL_TRACE      errors, warnings, notices & traces 
* 6 - LOG_LEVEL_VERBOSE    all 
```

Reply:
I see, notices could be like high-level messages whereas traces could be implementation details. For example, if we had a wheel speed sensor that communicates over I2C:

Trace: `I2C driver is sending 0x12...` 
Notice: `Speed exceeded 5m/s`. 

Just thinking out loud! Created an issue we can decide on later #25